src/csharp/Grpc.HealthCheck/HealthGrpc.cs

```diff
@@ -38,11 +38,22 @@ public interface IHealthClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IHealth
     {
       Task<global::Grpc.Health.V1.HealthCheckResponse> Check(global::Grpc.Health.V1.HealthCheckRequest request, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class HealthBase
```

Comment:
I'm surprised this doesn't implement the now-obsolete interface... any reason for that? (I would have thought that would make it easier for people to transition from one to the other.)